_source/_posts/2022-06-15-terraform-eks-microservices.md

```diff
@@ -0,0 +1,622 @@
+
+This doesn't mean that deploying and managing microservices on the public cloud is easy; they come with their own challenges and pain. This is especially true for Amazon EKS, which in my opinion, is the hardest to use Kubernetes service but also one of the most flexible. This is because EKS is nothing but some clever orchestrations doing a complex dance on top of other AWS services like EC2, EBS, etc.
+
+If you want to run a microservice stack on EKS, you would need to spend some extra time and effort setting it up and managing it. This is where infrastructure as code (IaC) tools like [Terraform](https://www.terraform.io/) comes in handy.
+
+So here is what you will learn to do today:
+
+- Scaffold a Java microservice stack using JHipster, Spring Boot, and Spring Cloud
+- Create an EKS cluster, Virtual Private Cloud (VPC), subnets, and required Kubernetes add-ons using Terraform on AWS
+- Set up OIDC authentication for the microservice stack using Okta
+- Build and deploy the microservice stack to the cluster
+
+**Prerequisites**
+
+- [AWS account](https://portal.aws.amazon.com/billing/signup) with the [IAM permissions to create EKS clusters](https://docs.aws.amazon.com/eks/latest/userguide/security_iam_id-based-policy-examples.html)
+- AWS CLI [installed](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) and [configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html)
+- [AWS IAM Authenticator](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html) installed
+- [kubectl](https://kubernetes.io/docs/tasks/tools/) installed
+- [Docker](https://docs.docker.com/get-docker/) installed and configured
+- [Terraform](https://www.terraform.io/downloads) installed
+- [Java 11+](https://sdkman.io/usage) installed
+- [Okta CLI](https://cli.okta.com/) installed
+- [Optional] [JHipster](https://www.jhipster.tech/installation/) CLI installed
+- [Optional] [KDash](https://github.com/kdash-rs/kdash)
+
+{% include toc.md %}
+
+## Why Terraform, why not CloudFormation?
+
+At this point, the first question that might pop up in your mind would be, "Why not use [CloudFormation](https://aws.amazon.com/cloudformation/)?". It's a good question; after all, CloudFormation is built by AWS and hence sounds like an excellent solution to manage AWS resources. But anyone who has tried both CloudFormation and Terraform would tell you to forget that CloudFormation even exists. I think CloudFormation is far more complex to use and less developer-friendly than Terraform. You also need to write a lot more boilerplate with CloudFormation in YAML or JSON. Yikes! And most importantly, Terraform is far more powerful and flexible than CloudFormation and is cross-platform, which means you can take care of all your infrastructure management needs on any platform with one tool.
```

Comment:
```suggestion
At this point, the first question that might pop up in your mind would be, "Why not use [CloudFormation](https://aws.amazon.com/cloudformation/)?". It's a good question; after all, CloudFormation is built by AWS and hence sounds like an excellent solution to manage AWS resources. But anyone who has tried both CloudFormation and Terraform will probably tell you to forget that CloudFormation even exists. I think CloudFormation is far more complex and less developer-friendly than Terraform. You also need to write a lot more boilerplate with CloudFormation in YAML or JSON. Yikes! And most importantly, Terraform is far more powerful and flexible than CloudFormation. It's cross-platform, which means you can take care of all your infrastructure management needs on any platform with one tool.
```
When making an opinionated unverifiable generalized statement ("anyone who has tried both CloudFormation and Terraform..."), it's worth qualifying your position. Just add "probably" or "most likely" or anything that acknowledges you don't have proof. I turned the closing sentence into two sentences, to give each statement more standalone impact. 

I t